components/Choose/index.tsx

```diff
@@ -39,6 +40,18 @@ export const Choose = () => {
     errors.unhandledError
   ])
 
+  const submitForm = useCallback(() => {
+    setSubmittedForm(true)
+    setMsigActor(safeID)
+  }, [setSubmittedForm, setMsigActor, safeID])
+
+  // Automatically submit the form if there is
+  // a valid MSIG address in the query params
+  useEffect(
+    () => msigAddressParam && isSafeIDValid && !submittedForm && submitForm(),
+    [msigAddressParam, isSafeIDValid, submittedForm, submitForm]
+  )
+
```

Comment:
@Schwartz10 this only works when the user also manually selects the right wallet account that was used to create the safe, or another wallet that is also a signer of the safe. Otherwise, the error will show that the wallet is not a signer.

If we want to prevent this, we might also want to pass the wallet address in the query parameters and use that to automatically select the right wallet in the previous screen as well.

However, it'd ben nice to skip the screens and do this in an automated step after logging in, so we don't see the interfaces flashing by

Reply:
@navFooh i think the following UX makes sense:

- If the user lands on the screen where they can enter or create a new actor with a msig address set in the query params, attempt to load the user into that msig without showing the enter or create actor screen (show a loading indicator)
- If the wrong wallet is selected, remove the query param, and display the enter or create actor screen (continue with normal flow)
- If the right wallet is selected, funnel them right into the dashboard